src/test2.py

```diff
@@ -249,8 +252,9 @@ def create_query(type: str, var_list, eq_list, goal_type):
 {var_names(var_list)} = {type_str(type)}['{var_names(var_list)}'].gens()
 gens = {eq_list}
 p = {goal_type}
-basis, coeffs, r = ideal_membership(p, gens)
-print(list(map(polynomial_to_string, coeffs[0])))
+I = ideal(gens)
```

Comment:
And, a big change snuck in at the end: this uses Sage's built in Grobner basis algorithm instead of your implementation. I wish we had known about this before you implemented your own version, but, oh well! I'm not sure it solves anything that yours doesn't...